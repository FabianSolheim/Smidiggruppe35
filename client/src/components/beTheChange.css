.image1 {
    width: 100%;
    background-image: url("../images/bethechangeimg1.png");
    background-repeat: no-repeat;
    background-size: auto;
    height: 400px;
}
.text1 {
    width: 100%;
    background-color: green;
    height: 400px;
}
.image2 {
    width: 100%;
    background-color: blue;
    height: 400px;
}
.text2 {
    width: 100%;
    background-color: yellow;
    height: 400px;
}

