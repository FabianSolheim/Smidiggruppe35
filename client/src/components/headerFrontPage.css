body{
    margin: 0;
}

.header {
    width: 100%;
    height: 600px;
    background-image: url("../images/frontPageBackground.png");
    background-repeat: no-repeat;
    background-size: cover;
    background-position: center;
    font-size: 64px;
}
.top {
    font-size: 0px;
}
.titleImage{
    width: 588px;
    height: 282px;
}
.description {
    font-size: 30px;
    font-family: "Montserrat", sans-serif;
    text-align: center;
}
.description em {
    color: #034F7A;
    font-style: normal;
}
.consultationBtn {
    margin-top: 80px;
    font-family: "Montserrat", sans-serif;
    border-radius: 0;
    height: 50px;
    width: 200px;
    background-color: #034F7A;
    color: white;
    border: 0;
    font-size: 16px;
}


@media only screen and (max-width: 750px) {

    .titleImage {
        width: 240px;
        height: 150px;
    }
    .headline {
        font-size: 34px;
        margin-top: 100px;
    }
    .description {
        font-size: 16px;
    }
    .consultationBtn {
        font-size: 16px;
        margin-top: 60px;
    }
    .header {
        max-height: 5em;
    }

}