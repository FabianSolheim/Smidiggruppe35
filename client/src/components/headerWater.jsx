import React from "react"
import "./header.css"

export function HeaderWater(){
    return <>
        <div className="header">
            <div>
                <p className="top">t</p>
                <p className="headline"><strong>Water</strong></p>
                <p className="description"><strong><em>785 million</em></strong> people in the world live <strong><em>without</em></strong><br/> clean water.</p>
            </div>
            <button className="consultationBtn"><strong>Book a Consultation</strong></button>
        </div>
    </>
}