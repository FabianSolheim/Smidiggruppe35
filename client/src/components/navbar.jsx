import {
    AppBar,
    Toolbar,
    Typography,
    Button,
    IconButton,
    Drawer,
    Grid,
    Link,
    Badge,
    MenuItem,
} from "@mui/material";
import { makeStyles } from "@mui/styles";
import MenuIcon from "@mui/icons-material/Menu";
import React, { useState, useEffect } from "react";
import { Link as RouterLink } from "react-router-dom";
import "../styling/style.css"

const headersData = [
    {
        title: "Home",
        href: "/",
    },
    {
        title: "About",
        href: "/about",
    },
    {
        title: "Contact",
        href: "/contact",
    },
    {
        title: "Privacy",
        href: "/privacy",
    },
    {
        title: "Log in",
        href: "/login",
    },
    {
        title: "Sign up",
        href: "/signup",
    },
];

const useStyles = makeStyles(() => ({
    header: {
        paddingRight: "79px",
        paddingLeft: "65px",
        "@media (max-width: 900px)": {
            paddingLeft: 0,
        },
    },

    menuButton: {
        color: "black",
        fontWeight: 700,
        size: "18px",
        marginRight: "30px",
        textTransform: "capitalize",
    },
    toolbar: {
        display: "flex",
        justifyContent: "space-between",
    },
    drawerContainer: {
        padding: "20px 30px",
    },

    paper: {
        justifyContent: "center",
        alignItems: "center",
        textAlign: "center",
        //width: 400,
        color: "#FFFFFF",
        backgroundColor: "#209b59",
    },
    typography: {
        fontFamily: "Montserrat",
    },
}));
export default function Header() {


    const classes = useStyles();

    const { header, menuButton, toolbar, drawerContainer, paper } =
        useStyles();

    const [state, setState] = useState({
        mobileView: false,
        drawerOpen: false,
    });

    const { mobileView, drawerOpen } = state;

    useEffect(() => {
        const setResponsiveness = () => {
            return window.innerWidth < 900
                ? setState((prevState) => ({ ...prevState, mobileView: true }))
                : setState((prevState) => ({ ...prevState, mobileView: false }));
        };

        setResponsiveness();

        window.addEventListener("resize", () => setResponsiveness());

        return () => {
            window.removeEventListener("resize", () => setResponsiveness());
        };
    }, []);

    const displayDesktop = () => {
        return (
            <Toolbar className={toolbar}>
                <a href="/"><p className="header-logo">Meliora <strong>impact</strong></p></a>
                <nav>
                    <ul>
                        {getMenuButtons()}
                    </ul>
                </nav>
            </Toolbar>
        );
    };

    const displayMobile = () => {
        const handleDrawerOpen = () =>
            setState((prevState) => ({ ...prevState, drawerOpen: true }));
        const handleDrawerClose = () =>
            setState((prevState) => ({ ...prevState, drawerOpen: false }));

        return (
            <AppBar style={{ backgroundColor: "rgba(255,238,238,0.2)", boxShadow: "none" }}>
                <Toolbar>

                        <h1 style={{ flexGrow: 1 }} className="header-logo">
                            <img src="https://i.postimg.cc/XYN8RB16/logo2.png"/>
                        </h1>
                    <IconButton
                        color="inherit"
                        aria-label="open drawer"
                        edge="end"
                        onClick={handleDrawerOpen}
                    >
                        <MenuIcon color="white" />
                    </IconButton>

                    <Drawer
                        anchor="right"
                        open={drawerOpen}
                        onClose={handleDrawerClose}
                        classes={{ paper: classes.paper }}
                    >
                        <div className={drawerContainer}>{getDrawerChoices()}</div>
                    </Drawer>
                </Toolbar>
            </AppBar>
        );
    };

    const getDrawerChoices = () => {
        return headersData.map(({ title, href }) => {
            return (
                <Link
                    {...{
                        component: RouterLink,
                        to: href,
                        color: "inherit",
                        style: { textDecoration: "none" },
                        key: title,
                    }}
                >
                    <MenuItem>{title}</MenuItem>
                </Link>
            );
        });
    };

    const getMenuButtons = () => {
        return (
            <>
                <Button
                    color="primary"
                    to="/"
                    component={RouterLink}
                    className={menuButton}
                    style={{fontSize: 18, fontFamily: "Montserrat", color: "white", textTransform: "capitalize"}}
                >
                    Home
                </Button>
                <Button
                    color="primary"
                    to="/about"
                    component={RouterLink}
                    className={menuButton}
                    style={{fontSize: 18, fontFamily: "Montserrat", color: "white", textTransform: "capitalize"}}
                >
                    About
                </Button>
                <Button
                    color="primary"
                    to="/contact"
                    component={RouterLink}
                    className={menuButton}
                    style={{fontSize: 18, fontFamily: "Montserrat", color: "white", textTransform: "capitalize"}}
                >
                    Contact
                </Button>
                <Button
                    color="primary"
                    to="/privacy"
                    component={RouterLink}
                    className={menuButton}
                    style={{fontSize: 18, fontFamily: "Montserrat", color: "white", textTransform: "capitalize"}}
                >

                    Privacy
                </Button>
                <Button
                    color="primary"
                    to="/login"
                    component={RouterLink}
                    className={menuButton}
                    style={{fontSize: 18, fontFamily: "Montserrat", color: "white", textTransform: "capitalize"}}
                >

                    Log in
                </Button>
                <Button
                    color="primary"
                    to="/Sign up"
                    component={RouterLink}
                    className={menuButton}
                    style={{fontSize: 18, fontFamily: "Montserrat", color: "white", textTransform: "capitalize"}}
                >

                    Sign up
                </Button>

            </>
        );
    };

    return (
        <header>
            <AppBar
                position="sticky"
                style={{backgroundColor: "rgba(255,238,238,0.2)", boxShadow: "none", marginBottom: -150}}
                className={header}
            >
                {mobileView ? displayMobile() : displayDesktop()}
            </AppBar>
        </header>
    );
}