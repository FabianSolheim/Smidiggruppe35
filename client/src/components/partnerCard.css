.partnerCard {
    text-align: center;
}
.centerDiv {
    margin: auto;
    width: 90%;
    background-color: #081825;
    padding: 10px;
    color: white;
}
.items {
    text-align: center;
}