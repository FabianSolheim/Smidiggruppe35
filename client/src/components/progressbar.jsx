import React, { useContext, useState } from "react";
import { Stepper, Step, StepLabel } from "@material-ui/core";

export function Progressbar({ activeStep }) {
  return (
    <div style={{ margin: 20 }}>
      <Stepper alternativeLabel activeStep={activeStep}>
        <Step>
          <StepLabel>Step 1: Select a date</StepLabel>
        </Step>
        <Step>
          <StepLabel>Step 2: Choose time</StepLabel>
        </Step>
        <Step>
          <StepLabel>Step 3: Fill the form</StepLabel>
        </Step>
        <Step>
          <StepLabel>All steps completed!</StepLabel>
        </Step>
      </Stepper>
    </div>
  );
}
