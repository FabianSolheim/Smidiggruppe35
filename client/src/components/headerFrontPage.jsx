import React from "react"

import React from "react"
import "./headerFrontPage.css"

export function HeaderFrontPage(){
    return <>
        <div className="header">
            <div>
                <p className="top">t</p>
                <img className="titleImage" src="https://i.postimg.cc/Kjwnys4K/Meliora-title.png"/>
                <p className="description">A digital platform where <strong><em>companies</em></strong><br/>
                    can give to nonprofits in a more<br/> <strong><em>impactful</em></strong>
                    and <strong><em>sustainable</em></strong> way</p>
            </div>
            <button className="consultationBtn"><strong>Book a Consultation</strong></button>
        </div>
    </>
}