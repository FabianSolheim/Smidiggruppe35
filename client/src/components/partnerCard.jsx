import React from "react"
import "./partnerCard.css"
import Box from '@mui/material/Box';
import Grid from '@mui/material/Grid';

export function PartnerCard(){
    return (
        <div className="centerDiv">
        <Box sx={{ flexGrow: 1 }}>
            <Grid container spacing={2}>
                <Grid item xs={4} md={4}>
                    <div className="items">Google</div>
                </Grid>
                <Grid item xs={4} md={4}>
                    <div className="items">HÃ¸yskolen Kristiania</div>
                </Grid>
                <Grid item xs={4} md={4}>
                    <div className="items">Linkedin</div>
                </Grid>
            </Grid>
        </Box>
        </div>
    );
}