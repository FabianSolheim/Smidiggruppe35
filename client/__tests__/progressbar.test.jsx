import React from "react";
import ReactDOM from "react-dom";
import { Progressbar } from "../src/components/progressbar";

describe("ProgressBar test", () => {
  it("shows progressBar", () => {
    const domElement = document.createElement("div");
    ReactDOM.render(<Progressbar />, domElement);
    expect(domElement.innerHTML).toMatchSnapshot();
  });
});
