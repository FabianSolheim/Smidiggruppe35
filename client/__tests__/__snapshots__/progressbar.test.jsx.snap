// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar test shows progressBar 1`] = `"<div style=\\"margin: 20px;\\" class=\\"makeStyles-root-1\\"><div class=\\"MuiPaper-root MuiStepper-root MuiStepper-horizontal MuiStepper-alternativeLabel MuiPaper-elevation0\\"><div class=\\"MuiStep-root MuiStep-horizontal MuiStep-alternativeLabel\\"><span class=\\"MuiStepLabel-root MuiStepLabel-horizontal MuiStepLabel-alternativeLabel\\"><span class=\\"MuiStepLabel-iconContainer MuiStepLabel-alternativeLabel\\"><svg class=\\"MuiSvgIcon-root MuiStepIcon-root MuiStepIcon-active\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><circle cx=\\"12\\" cy=\\"12\\" r=\\"12\\"></circle><text class=\\"MuiStepIcon-text\\" x=\\"12\\" y=\\"16\\" text-anchor=\\"middle\\">1</text></svg></span><span class=\\"MuiStepLabel-labelContainer\\"><span class=\\"MuiTypography-root MuiStepLabel-label MuiStepLabel-alternativeLabel MuiStepLabel-active MuiTypography-body2 MuiTypography-displayBlock\\">Step 1: Select a date</span></span></span></div><div class=\\"MuiStep-root MuiStep-horizontal MuiStep-alternativeLabel\\"><div class=\\"MuiStepConnector-root MuiStepConnector-horizontal MuiStepConnector-alternativeLabel Mui-disabled\\"><span class=\\"MuiStepConnector-line MuiStepConnector-lineHorizontal\\"></span></div><span class=\\"MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled MuiStepLabel-alternativeLabel\\"><span class=\\"MuiStepLabel-iconContainer MuiStepLabel-alternativeLabel\\"><svg class=\\"MuiSvgIcon-root MuiStepIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><circle cx=\\"12\\" cy=\\"12\\" r=\\"12\\"></circle><text class=\\"MuiStepIcon-text\\" x=\\"12\\" y=\\"16\\" text-anchor=\\"middle\\">2</text></svg></span><span class=\\"MuiStepLabel-labelContainer\\"><span class=\\"MuiTypography-root MuiStepLabel-label MuiStepLabel-alternativeLabel MuiTypography-body2 MuiTypography-displayBlock\\">Step 2: Choose time</span></span></span></div><div class=\\"MuiStep-root MuiStep-horizontal MuiStep-alternativeLabel\\"><div class=\\"MuiStepConnector-root MuiStepConnector-horizontal MuiStepConnector-alternativeLabel Mui-disabled\\"><span class=\\"MuiStepConnector-line MuiStepConnector-lineHorizontal\\"></span></div><span class=\\"MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled MuiStepLabel-alternativeLabel\\"><span class=\\"MuiStepLabel-iconContainer MuiStepLabel-alternativeLabel\\"><svg class=\\"MuiSvgIcon-root MuiStepIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><circle cx=\\"12\\" cy=\\"12\\" r=\\"12\\"></circle><text class=\\"MuiStepIcon-text\\" x=\\"12\\" y=\\"16\\" text-anchor=\\"middle\\">3</text></svg></span><span class=\\"MuiStepLabel-labelContainer\\"><span class=\\"MuiTypography-root MuiStepLabel-label MuiStepLabel-alternativeLabel MuiTypography-body2 MuiTypography-displayBlock\\">Step 3: Fill the form</span></span></span></div><div class=\\"MuiStep-root MuiStep-horizontal MuiStep-alternativeLabel\\"><div class=\\"MuiStepConnector-root MuiStepConnector-horizontal MuiStepConnector-alternativeLabel Mui-disabled\\"><span class=\\"MuiStepConnector-line MuiStepConnector-lineHorizontal\\"></span></div><span class=\\"MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled MuiStepLabel-alternativeLabel\\"><span class=\\"MuiStepLabel-iconContainer MuiStepLabel-alternativeLabel\\"><svg class=\\"MuiSvgIcon-root MuiStepIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><circle cx=\\"12\\" cy=\\"12\\" r=\\"12\\"></circle><text class=\\"MuiStepIcon-text\\" x=\\"12\\" y=\\"16\\" text-anchor=\\"middle\\">4</text></svg></span><span class=\\"MuiStepLabel-labelContainer\\"><span class=\\"MuiTypography-root MuiStepLabel-label MuiStepLabel-alternativeLabel MuiTypography-body2 MuiTypography-displayBlock\\">All steps completed!</span></span></span></div></div></div>"`;
